packages/serverless-components/nextjs-cdk-construct/__tests__/construct.test.ts

```diff
@@ -0,0 +1,169 @@
+import "@aws-cdk/assert/jest";
+import { SynthUtils } from "@aws-cdk/assert";
+import { Stack } from "@aws-cdk/core";
+import path from "path";
+import { NextJSLambdaEdge } from "../src";
+import { Runtime } from "@aws-cdk/aws-lambda";
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import { HostedZone } from "@aws-cdk/aws-route53";
+
+describe("CDK Construct", () => {
+  it("passes correct lambda options to underlying lambdas when single value passed", () => {
+    const stack = new Stack();
+    new NextJSLambdaEdge(stack, "Stack", {
+      serverlessBuildOutDir: path.join(__dirname, "fixtures/next-boilerplate"),
+      runtime: Runtime.NODEJS_10_X
+    });
+
+    const synthesizedStack = SynthUtils.toCloudFormation(stack);
+    expect(synthesizedStack).toHaveResourceLike("AWS::Lambda::Function", {
+      FunctionName: "NextDefaultLambda",
+      Runtime: Runtime.NODEJS_10_X.name
+    });
+    expect(synthesizedStack).toHaveResourceLike("AWS::Lambda::Function", {
+      FunctionName: "NextApiLambda",
+      Runtime: Runtime.NODEJS_10_X.name
+    });
+    expect(synthesizedStack).toHaveResourceLike("AWS::Lambda::Function", {
+      FunctionName: "NextImageLambda",
+      Runtime: Runtime.NODEJS_10_X.name
+    });
+  });
+
+  it("passes correct lambda options to underlying lambdas when object passed", () => {
+    const stack = new Stack();
+    new NextJSLambdaEdge(stack, "Stack", {
+      serverlessBuildOutDir: path.join(__dirname, "fixtures/next-boilerplate"),
+      runtime: {
+        defaultLambda: Runtime.PYTHON_3_8,
```

Comment:
Not sure why we are using Python/Go/Java runtimes here. Is it just to test that you can pass different runtimes for unit tests?

Reply:
Correct, i thought this might look a lil strange. Yes, this test just asserts that the runtime prop correctly passes through to the CF template.